week05/store_test.go

```diff
@@ -0,0 +1,147 @@
+package demo
+
+import (
+	"errors"
+	"testing"
+)
+
+func Test_Store(t *testing.T) {
+	t.Parallel()
+
+	tn := "test_table"
+	tk := "test_key"
+	tv := "test_val"
+	mod := Model{tk: tv}
+
+	emptyStore := &Store{}
+	withEmptyData := &Store{data: data{}}
+	withEmptyTable := &Store{data: data{tn: Models{}}}
+
+	withPopulatedStore := &Store{}
+	withPopulatedStore.Insert(tn, mod)
+
+	t.Run("Len", func(t *testing.T) {
+		t.Parallel()
+
+		table := []struct {
+			name    string
+			store   *Store
+			query   string
+			exp_len int
```

Comment:
In Go we use Camel/PascalCase for naming. 

```suggestion
			expLen int
```